### PR TITLE
Recentrifuge update for html summary

### PR DESCRIPTION
**Pull request by pimarin**
Updating the recentrifuge tool for :

- edam number
- little option in the html output which render 1 html file with all samples

### CODE REVIEW

File: tools/recentrifuge/macro.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <macros>
   <token name="@TOOL_VERSION@">1.9.1</token>
-  <token name="@VERSION_SUFFIX@">0</token>
+  <token name="@VERSION_SUFFIX@">0.1</token>
```
Comment (bgruening) [inline tools/recentrifuge/macro.xml:4]: The version suffix can be a full number. Just use 1 here.

File: tools/recentrifuge/recentrifuge.xml [hunk 2]
```diff
@@ -88,7 +107,7 @@
     </command>
     <inputs>
         <!-- INPUT FILES -->
-        <param name="input_file" type="data" format="tabular" label="Select taxonomy file tabular formated"/>
+        <param name="input_file" type="data" format="tabular" multiple="true" optional="true" label="Select taxonomy file tabular formated"/>
```
Comment (bgruening) [inline tools/recentrifuge/recentrifuge.xml:110]: Isn't that your main I put file? Why can be optional=true?

File: tools/recentrifuge/macro.xml [hunk 3]
```diff
@@ -16,6 +16,16 @@
       <requirement type="package" version="@TOOL_VERSION@">recentrifuge</requirement>
     </requirements>
   </xml>
+  <xml name="edam">
+  <edam_topics>
```
Comment (bernt-matthias) [inline tools/recentrifuge/macro.xml:20]: Please indent the macro content

File: tools/recentrifuge/macro.xml (hunk 1 above)
Comment (pimarin) [reply tools/recentrifuge/macro.xml:4]: changed to 1

File: tools/recentrifuge/recentrifuge.xml (hunk 2 above)
Comment (pimarin) [reply tools/recentrifuge/recentrifuge.xml:110]: I made a mistake between optional for multiple, I changed it

File: tools/recentrifuge/macro.xml (hunk 3 above)
Comment (pimarin) [reply tools/recentrifuge/macro.xml:20]: checked

**Comment by pimarin** [comment]
I pushed the last commit with change of the version number, indentation error and optional="false"

File: tools/recentrifuge/recentrifuge.xml [hunk 4]
```diff
@@ -7,10 +7,27 @@
...
     <expand macro="requirements" />
     <expand macro="version_command" />
     <command detect_errors="aggressive"><![CDATA[
+      #*======================================
+            Make a dir with all input files
+      ======================================*#
+        mkdir input_dir &&
+        #for $input in $input_file
+            #if $file_type.filetype == "centrifuge"
+                ln -s '$input' 'input_dir/$input.element_identifier''.out' &&
```
Comment (bernt-matthias) [inline tools/recentrifuge/recentrifuge.xml:21]: The `element_identifier` can contain spaces and all sorts of special characters. Its usually better to replace them with something safe like `_`. 

See here https://github.com/galaxyproject/tools-iuc/blob/586b4d3a59a38c34b5052359810a8e8d589830d6/tools/mykrobe/mykrobe_predict.xml#L33

File: tools/recentrifuge/recentrifuge.xml [hunk 5]
```diff
@@ -19,23 +36,25 @@
...
             #set $default_scoring = 'SHEL'
         #else if $file_type.filetype == "kraken"
-            -k '$input_file'
+            -k input_dir
             #set $default_scoring = 'KRAKEN'
+        #else if $file_type.filetype == "clark"
+            -r input_dir
+            #set $default_scoring = 'SHEL'
+        #else if $file_type.filetype == "lmat"
+            -l '$input_type.input_file'
```
Comment (bernt-matthias) [inline tools/recentrifuge/recentrifuge.xml:50]: `'$input_type.input_file'` seems not to exist .. here and in the `generic` branch. 

Can you add a test for each of the input types?

Comment (pimarin) [reply tools/recentrifuge/recentrifuge.xml:50]: I failed to recover file from LMAT and generic taxonomic tools, impossible to install to install the lmat tools

File: tools/recentrifuge/recentrifuge.xml (hunk 4 above)
Comment (pimarin) [reply tools/recentrifuge/recentrifuge.xml]: I created a ` #set input_identifier = re.sub('[^\s\w\-]', '_', str($input.element_identifier)) `

File: tools/recentrifuge/recentrifuge.xml (hunk 5 above)
Comment (pimarin) [reply tools/recentrifuge/recentrifuge.xml]: I'm trying to install CLARK and test it

File: tools/recentrifuge/recentrifuge.xml [hunk 6]
```diff
@@ -7,10 +7,26 @@
...
     <expand macro="version_command" />
     <command detect_errors="aggressive"><![CDATA[
+      #*======================================
+            Make a dir with all input files
+      ======================================*#
+        mkdir input_dir &&
+        #for $input in $input_file
+            #set input_identifier = re.sub('[^\s\w\-]', '_', str($input.element_identifier))
+            #if $file_type.filetype == "centrifuge"
+                ln -s '$input' 'input_dir/$input_identifier''.out' &&
```
Comment (bernt-matthias) [inline tools/recentrifuge/recentrifuge.xml:22]: You can also simplify that with

```
                ln -s '$input' 'input_dir/${input_identifier}.out' &&
```

Comment (pimarin) [reply tools/recentrifuge/recentrifuge.xml]: I modified it but removing `&&`  failed

Comment (bernt-matthias) [reply tools/recentrifuge/recentrifuge.xml]: That's fine. `&&` needs to stay.

File: tools/recentrifuge/recentrifuge.xml (hunk 5 above)
Comment (pimarin) [reply tools/recentrifuge/recentrifuge.xml]: I failed to use CLARK after trying to install by conda or using the tar.gz

File: tools/recentrifuge/recentrifuge.xml [hunk 7]
```diff
@@ -7,10 +7,28 @@
...
+            #set input_identifier = re.sub('[^\s\w\-]', '_', str($input.element_identifier))
+            echo $input_identifier &&
+            #if $file_type.filetype == "centrifuge"
+                ln -s '$input' 'input_dir/${input_identifier}.out' &&
+            #else if $file_type.filetype == "clark"
+                ln -s '$input' 'input_dir/${input_identifier}.csv' &&
+            #else if $file_type.filetype == "kraken"
+                ln -s '$input' 'input_dir/${input_identifier}.krk' &&
+            #end if
+            ln -s '$input' 'input_dir/$input_identifier' &&
```
Comment (bgruening) [inline tools/recentrifuge/recentrifuge.xml:30]: Why is this $input linked twice?

File: tools/recentrifuge/recentrifuge.xml [hunk 8]
```diff
@@ -7,10 +7,28 @@
...
     <expand macro="version_command" />
     <command detect_errors="aggressive"><![CDATA[
+      #import re
+      #*======================================
+            Make a dir with all input files
+      ======================================*#
+        mkdir input_dir &&
+        #for $input in $input_file
+            #set input_identifier = re.sub('[^\s\w\-]', '_', str($input.element_identifier))
+            echo $input_identifier &&
```
Comment (bgruening) [inline tools/recentrifuge/recentrifuge.xml:22]: this echo is not needed isn't it?

**Review by bgruening — COMMENTED** [review]
* $logfile is not quoted

And should we output the logfile only optionally?

**Comment by pimarin** [comment]
This PR is okay ?

**Review by bernt-matthias — APPROVED** [review]

**Review by bgruening — APPROVED** [review]

**Pull request merged**